src/libcore/iter/traits/iterator.rs

```diff
@@ -2541,11 +2642,9 @@ pub trait Iterator {
         };
 
         while let Some(curr) = self.next() {
-            if compare(&last, &curr)
-                .map(|o| o == Ordering::Greater)
-                .unwrap_or(true)
-            {
-                return false;
+            match compare(&last, &curr) {
```

Comment:
Use `if let`?

Reply:
Does `if let X | Y` work now? If so, sure!

Reply:
It does :) (https://github.com/rust-lang/rust/pull/57532)

Reply:
Oh wow, way back in 1.33! :sweat_smile:

Reply:
@cuviper And you are supposedly on the release team... =P

Reply:
touché